contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -924,6 +924,38 @@ There are multiple scenarios where a desired resource may not exist. Examples in
 Controllers should be authored with the assumption that the referenced resource may not exist, and include
 error handling to make the issue clear to the user.
 
+### Validation of fields
+
+Many of the values used in an object reference are used as part of the API path. For example,
+the object name is used in the path to identify the object. Unsanitized, these values can be used to
+attempt to retrieve other resources, such as by using values with semantic meanings such as  `..` or `/`.
+
+Have the controller validate the field before using it as a reference, and emit an event to
+tell the user that the validation has failed.
+
+See [Object Names and IDs](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names)
```

Comment:
link to the overall name topic which describes three common types of name validation:
```suggestion
See [Object Names and IDs](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/)
```